creator-node/src/snapbackSM/secondarySyncHealthTracker.js

```diff
@@ -19,10 +19,17 @@ const Outcomes = Object.freeze({
 const Utils = {
   /**
    * Given redis key pattern, returns all keys matching pattern and associated values
-   * Returns map of key-value pairs
+   *
+   * @param {string} pattern the pattern to run a redis SCAN on to find matching keys
+   * @param {String[]} wallets optional array of wallets to filter matched keys by
+   * @returns map of key-value pairs
    */
-  async _getMetricsMatchingPattern(pattern) {
-    const keys = await Utils._getAllKeysMatchingPattern(pattern)
+  async _getMetricsMatchingPattern(pattern, wallets = []) {
+    const keys = wallets
```

Comment:
ah i see what's happening
previously we could only search for either all wallets or a single wallet in the pattern
i can see how this scan is pretty not great haha
we're basically scanning every single wallet and then filtering in memory right

maybe we can talk about a better way to do this, with eg redis pipeline? may not be highest pri tho

Reply:
yeah it scans in batches of 10 to not block like KEYS does, but still it has to search through the whole db so not ideal. we have [this ticket](https://linear.app/audius/issue/CON-135/refactor-sync-metrics-to-avoid-using-redis-scan) so we'll make sure it doesn't slip through the cracks

Reply:
perf